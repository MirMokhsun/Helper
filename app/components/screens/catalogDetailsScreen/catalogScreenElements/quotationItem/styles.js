import { StyleSheet, I18nManager } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 20,
        justifyContent: 'space-between'
    },
    text: {
        fontFamily: 'Roboto-Light',
        lineHeight: 20,
        fontSize: 15,
        color: '#000756',
    },
    textWrapper:{
        flex: 2,
        justifyContent: 'flex-end',
        paddingRight: 10,
        flexDirection: 'row',
    },
    rendersNumbWrapper: {
        flexDirection: I18nManager.isRTL ? 'row-reverse' : 'row',
        justifyContent: 'flex-start',
        flex: 3
    },
    whiteNumbers: {
        lineHeight: 20,
        fontSize: 15,
        color: "#9FA0B2",
        fontFamily: 'Roboto-Light',
    },
    largeNumbers: {
        paddingTop: 0.5,
        fontSize: 18,
    },
    valuestyle: {

    },
    redNumbers: {
        lineHeight: 20,
        fontSize: 15,
        color: "#F86F6F",
        fontFamily: 'Roboto-Light',
    },
    greenNumbers: {
        lineHeight: 20,
        fontSize: 15,
        color: "#1459D2",
        fontFamily: 'Roboto-Light',
    },
})