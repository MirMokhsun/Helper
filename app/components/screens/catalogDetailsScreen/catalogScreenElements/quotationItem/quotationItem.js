import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles';
import { isEqualNumb, setCamelPos } from '../../../../../api/cardQuotation.api/cardquotation.api';

export class QuotationItem extends React.PureComponent {
    constructor(props) {
        super(props)
        this.countUperNambers = 2;
        this.rendersNumb = setCamelPos(props.number, styles.whiteNumbers, this.countUperNambers, styles);
        this.state = {
            rendersNumb: this.rendersNumb,
        }
    }

    UNSAFE_componentWillReceiveProps({ number }) {
        const { number: thisNumber } = this.props;
        if (thisNumber !== number) {
            const compare = isEqualNumb(thisNumber, number);
            const colorNumbers = compare ? styles.greenNumbers : styles.redNumbers;
            const rendersNumb = setCamelPos(number, colorNumbers, this.countUperNambers, styles);
            this.setState({
                rendersNumb,
            });
        }
    }

    render() {
        const { text } = this.props;
        const { rendersNumb } = this.state;

        return (
            <View style={styles.container}>
                <View style={styles.textWrapper}>
                    <Text style={styles.text}>{text}</Text>
                </View>
                <View style={styles.rendersNumbWrapper}>
                    {rendersNumb}
                </View>
            </View >
        )
    }
}

export default QuotationItem;