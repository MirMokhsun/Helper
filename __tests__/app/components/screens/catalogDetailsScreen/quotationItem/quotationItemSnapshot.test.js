import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { QuotationItem } from '../../../../../../app/components/screens/catalogDetailsScreen/catalogScreenElements/quotationItem/quotationItem';

it('create snapshot QuotationItem', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<QuotationItem text={'local.cellItem.titleOfBuy'} number={"123456"} />);
    expect(toolbarSnapshot).toMatchSnapshot();
});
