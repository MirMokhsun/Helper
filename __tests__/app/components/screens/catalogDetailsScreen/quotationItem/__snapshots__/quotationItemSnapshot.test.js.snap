// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot QuotationItem 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 20,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000756",
          "fontFamily": "Roboto-Light",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
    >
      local.cellItem.titleOfBuy
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9FA0B2",
            "fontFamily": "Roboto-Light",
            "fontSize": 15,
            "lineHeight": 20,
          },
          Object {},
        ]
      }
    >
      1
      2
      3
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9FA0B2",
            "fontFamily": "Roboto-Light",
            "fontSize": 15,
            "lineHeight": 20,
          },
          Object {
            "fontSize": 18,
            "paddingTop": 0.5,
          },
        ]
      }
    >
      4
      5
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#9FA0B2",
            "fontFamily": "Roboto-Light",
            "fontSize": 15,
            "lineHeight": 20,
          },
          Object {},
        ]
      }
    >
      6
    </Text>
  </View>
</View>
`;
